src/server/actions/user-actions.test.ts

```diff
@@ -0,0 +1,309 @@
+import { describe, it, expect, vi, afterEach } from 'vitest'
+import { db } from '@/database'
+import { getUserIdByClerkId, getMemberUserPublicKey, findOrCreateSiUserId, createUserAction } from './user-actions'
+
+// Mock the database
+vi.mock('@/database', () => ({
+    db: {
+        selectFrom: vi.fn(),
+        insertInto: vi.fn(),
+        transaction: vi.fn(),
+    },
+}))
```

Comment:
I don't think we should mock the DB unless we really need to.  we're using `pg-transactional-tests` library which runs everything inside a transaction that's rolled back at the end of each spec.